06week/higherOrder.js

```diff
@@ -2,102 +2,145 @@
 
 const assert = require('assert');
 
+
 function forEach(arr, callback) {
-  // Your code here
+    // new array with converted items
+    let result = [];
+
+    // loop through entire length of array
+    for (let i = 0; i < arr.length; i++) {
+        result.push(callback(arr[i]));
+    }
+
+    // return new array with converted elements
+    return result;
 }
 
 function map(arr, callback) {
-  // Your code here
+    // new array with mapped elements
+    let result = [];
+
+    // loop through array and convert element
+    for (let i = 0; i < arr.length; i++) {
+        result.push(callback(arr[i]));
+    }
+
+    // return new array with mapped elements
+    return result;
 }
 
 function filter(arr, callback) {
-  // Your code here
+    // new array with filtered elements
+    let result = [];
+
+    // loop through array and filter element
+    for (let i = 0; i < arr.length; i++) {
+        if (arr[i] % 2 === 0) {
```

Comment:
this is not applying the callback to every element. What you need is
```
function filter(arr, callback) {
    // new array with filtered elements
    let result = [];

    // loop through array and filter element
    for (let i = 0; i < arr.length; i++) {
        if (callback(arr[i])) {
            result.push(arr[i]);
        };
    }
    // return new array with filtered elements
    return result;
}
```

Notice the difference between my if condition and yours.